cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/AdaptTemplatesCommand.java

```diff
@@ -13,10 +24,103 @@
 "a" }, mixinStandardHelpOptions = true)
 public class AdaptTemplatesCommand extends CommandCommons {
 
+  /**
+   * Logger to output useful information to the user
+   */
+  private static Logger LOG = LoggerFactory.getLogger(CobiGenCLI.class);
+
   @Override
   public Integer doAction() throws Exception {
 
-    CobiGenFactory.extractTemplates();
+    TemplateAdapter templateAdapter;
+    if (this.templatesProject == null) {
+      templateAdapter = new TemplateAdapterImpl();
+    } else {
+      templateAdapter = new TemplateAdapterImpl(this.templatesProject);
+    }
+
+    if (templateAdapter.isMonolithicTemplatesConfiguration()) {
+      Path destinationPath = templateAdapter.getTemplatesLocation().resolve(ConfigurationConstants.COBIGEN_TEMPLATES);
+      templateAdapter.adaptMonolithicTemplates(destinationPath, false);
+
+      LOG.info("Templates were adapted to {}", destinationPath);
+      if (askUserToContinueWithUpgrade()) {
+        templateAdapter.upgradeMonolithicTemplates();
+      }
+    } else {
+      List<Path> templateJars = templateAdapter.getTemplateSetJars();
+      if (templateJars != null && !templateJars.isEmpty()) {
+        List<Path> templateJarsToAdapt = getJarsToAdapt(templateAdapter, templateJars);
+        if (!templateJarsToAdapt.isEmpty()) {
+          templateAdapter.adaptTemplateSets(templateJarsToAdapt, false);
+        }
+      } else {
+        LOG.info("No template set jars found to extract.");
+      }
+    }
```

Comment:
Couldn't most of this logic be moved to the core f.e. cobigenFactory?

Reply:
What exactly should be moved?
Most of the logic is contained in the TemplateAdapter itself. Methods like `askUserToContinueWithUpgrade` or `getJarsToAdapt` should be in the cli module as they use input from the console. This implementation is anyway just for now and will be replaced by the own UI that core will provide in the future.

Reply:
The TemplateAdapter is part of core right?
Why should the cli know about the configuration to be monolithic or not? I agree that the whole case decision and action could easily be coupled into core. Otherwise you have to implement it at each consumer again.